--- Core/Src/freertos.c
+++ Core/Src/freertos.c
@@ -27,6 +27,7 @@
 /* Private includes
  * ----------------------------------------------------------*/
 /* USER CODE BEGIN Includes */
+#include "app/app.hpp"
 
 /* USER CODE END Includes */
 
@@ -57,7 +58,7 @@
 osThreadId_t defaultTaskHandle;
 const osThreadAttr_t defaultTask_attributes = {
     .name = "defaultTask",
-    .stack_size = 128 * 4,
+    .stack_size = 256 * 4,
     .priority = (osPriority_t)osPriorityNormal,
 };
 
@@ -120,9 +121,9 @@
 /* USER CODE END Header_StartDefaultTask */
 void StartDefaultTask(void *argument) {
     /* USER CODE BEGIN StartDefaultTask */
-    /* Infinite loop */
+    (void)argument;
     for (;;) {
-        osDelay(1);
+        osDelay(portMAX_DELAY);
     }
     /* USER CODE END StartDefaultTask */
 }
