--- Core/Src/gpio.c
+++ Core/Src/gpio.c
@@ -59,9 +59,9 @@
     /*Configure GPIO pin Output Level */
     HAL_GPIO_WritePin(SD_CS_GPIO_Port, SD_CS_Pin, GPIO_PIN_RESET);
 
-    /*Configure GPIO pin : USR_BTN_Pin */
+    /*Configure GPIO pin : USR_BTN_Pin (EXTI for button_handler) */
     GPIO_InitStruct.Pin = USR_BTN_Pin;
-    GPIO_InitStruct.Mode = GPIO_MODE_INPUT;
+    GPIO_InitStruct.Mode = GPIO_MODE_IT_RISING_FALLING;
     GPIO_InitStruct.Pull = GPIO_NOPULL;
     HAL_GPIO_Init(USR_BTN_GPIO_Port, &GPIO_InitStruct);
 
@@ -86,6 +86,9 @@
     GPIO_InitStruct.Pull = GPIO_NOPULL;
     GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
     HAL_GPIO_Init(SD_CS_GPIO_Port, &GPIO_InitStruct);
+
+    HAL_NVIC_SetPriority(EXTI0_IRQn, 5, 0);
+    HAL_NVIC_EnableIRQ(EXTI0_IRQn);
 }
 
 /* USER CODE BEGIN 2 */
